github/orgs/retrieve_github_data.py

```diff
@@ -0,0 +1,364 @@
+    )
+    ap.add_argument(
+        "--no-csv",
+        help="Do not output to CSV (default False if called via cli).",
+        action="store_true",
+    )
+    ap.add_argument("--no-json", help="Do not output JSON.", action="store_true")
+    ap.add_argument(
+        "--json",
+        help="JSON output file name (default 'org.json')",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+    )
+    meg = ap.add_mutually_exclusive_group()
+    meg.add_argument("--prod", help="run against prod org set", action="store_true")
+    meg.add_argument("--all", help="run against all orgs you own", action="store_true")
+
+    ap.add_argument(
+        "orgs", nargs="*", help='Organization slug name, such as "mozilla".'
+    )
+
+    args = ap.parse_args()
+
+    endpoint_loglevel = max(10, 40 - ((args.verbose - 3) * 10))
+    logfmt = "%(levelname)s: %(message)s"
+    if endpoint_loglevel < logging.ERROR:
+        logfmt = "%(levelname)s:%(name)s: %(message)s"
+
+    logging.basicConfig(format=logfmt, level=max(10, 40 - (args.verbose * 10)))
+    HTTPEndpoint.logger.setLevel(endpoint_loglevel)
+
+    if not args.token:
+        ap.error(
+            "token must be provided. You may create an "
+            "app or personal token at "
+            "https://github.com/settings/tokens"
+        )
+    if not (args.prod or args.all) and (len(args.orgs) == 0):
+        ap.error("Must supply 'repo' or set either '--prod' or '--all'")
+    return args
+
+
+def _in_offline_mode() -> bool:
+    is_offline = False
+    try:
+        # if we're running under pytest, we need to fetch the value from
+        # the current configuration
+        import conftest
+
+        is_offline = conftest.get_client("github_client").is_offline()
+        if not is_offline:
+            # check for a valid GH_TOKEN here so we fail during test collection
+            # TODO: make sure this works in all scenarios
+            os.environ["GH_TOKEN"]
+
+    except ImportError:
+        pass
+
+    return is_offline
+
+
+def _orgs_to_check() -> Set[str]:
+    # just shell out for now
+    #   While there is no network operation done here, we don't want to go
+    #   poking around the file system if we're in "--offline" mode
+    #   (aka doctest mode)
+    if _in_offline_mode():
+        return []
+    # DEV_HACK: find better way to insert dev default
+    path_to_metadata = os.environ.get(
+        "PATH_TO_METADATA", "~/repos/foxsec/master/services/metadata"
+    )
+    meta_dir = Path(os.path.expanduser(path_to_metadata)).resolve()
+    in_files = list(meta_dir.glob("*.json"))
+
+    cmd = [
+        "jq",
+        "-rc",
+        """.codeRepositories[]
+                | select(.status == "active")
+                | [.url]
+                | @csv
+                """,
+        *in_files,
+    ]
+    # python 3.6 doesn't support capture_output
+    status = subprocess.run(cmd, capture_output=True)  # nosec
```

Comment:
Why not just read the json files and parse them within Python?

Reply:
because 'jq' is more expressive, and easier to change? (<== my reasons during development)

I'd prefer to move to "[jq bindings](https://pypi.org/project/jq/)", and get rid of the "`subprocess`" ugliness, unless that still raises the same concerns.

@ajvb thoughts?

Reply:
I'd definitely prefer to use the python std lib over requiring another dependency, especially an additional system package. As well, though it is more expressive, it requires any new developer to learn how to use `jq` without providing what seems to me to be enough value to justify that.

Reply:
Yeah - there seems to be a great debate on whether "reading jq" is required skill, and whether "jq" should be a standard tool in images.  I vote "yes" on both, but ulfr switched ("yes" when I first started, then "meh" later)

I'll rewrite